plugins/modules/inventory_workflow_manager.py

```diff
@@ -1870,25 +2174,42 @@ def get_site_type(self, site_name):
         """
 
         try:
-            site_type = None
-            response = self.dnac_apply['exec'](
-                family="sites",
-                function='get_site',
-                params={"name": site_name},
-            )
+            if self.get_ccc_version_as_integer() <= self.get_ccc_version_as_int_from_str("2.3.5.3"):
+                site_type = None
+                response = response = self.get_site(site_name)
 
-            if not response:
-                self.msg = "Site '{0}' not found".format(site_name)
-                self.log(self.msg, "INFO")
-                return site_type
+                self.log("Received API response from 'get_site': {0}".format(str(response)), "DEBUG")
 
-            self.log("Received API response from 'get_site': {0}".format(str(response)), "DEBUG")
-            site = response.get("response")
-            site_additional_info = site[0].get("additionalInfo")
+                if not response:
+                    self.msg = "Site '{0}' not found".format(site_name)
+                    self.log(self.msg, "INFO")
+                    return site_type
 
-            for item in site_additional_info:
-                if item["nameSpace"] == "Location":
-                    site_type = item.get("attributes").get("type")
+                site = response.get("response")
+                site_additional_info = site[0].get("additionalInfo")
+
+                for item in site_additional_info:
+                    if item["nameSpace"] == "Location":
+                        site_type = item.get("attributes").get("type")
+
+        except Exception as e:
+            self.msg = "Error while fetching the site '{0}' and the specified site was not found in Cisco Catalyst Center.".format(site_name)
+            self.log(self.msg, "ERROR")
+            self.module.fail_json(msg=self.msg, response=[self.msg])
+
+        try:
+            if self.get_ccc_version_as_integer() >= self.get_ccc_version_as_int_from_str("2.3.7.6"):
+                site_type = None
+                response = self.get_site(site_name)
+
+                if not response:
+                    self.msg = "Site '{0}' not found".format(site_name)
+                    self.log(self.msg, "INFO")
+                    return site_type
+
+                self.log("Received API response from 'get_sites': {0}".format(str(response)), "DEBUG")
+                site = response.get("response")
```

Comment:
Add a check if you receive site response or not .

Reply:
already we have checked above